### ISSUE TITLE
Travis tests failing

### ISSUE THREAD

**Issue by lukehoban** [pulumi/pulumi-eks#300]
Travis tests have failed or errored for the last 16 days, and for 20 of the 21 latest runs in `master`.

**Comment by metral** [comment]
I've combed through the last handful of failed CI tests on master, and the errors are the same as reported in #267:

```
Error: 2 UNKNOWN: invocation of aws:ec2/getRouteTable:getRouteTable returned an error:
invoking aws:ec2/getRouteTable:getRouteTable:
Your query returned no results. Please change your search criteria and try again, resp: undefined
```

We're seeing this error across all examples/tests that have at least 1 cluster defined that depends on the account's default VPC to be retrieved and used.

- `examples/cluster`
- `examples/nodegroup`
- `examples/nodegroup-options`
- `examples/managed-nodegroups`
- `examples/storage-classes`
- `examples/tags`
- `examples/tag-input-types`

Related PR that attempted to fix #267: #264

cc @CyrusNajmabadi

**Comment by metral** [comment]
cc @lukehoban

**Comment by metral** [comment]
Ran a slew of attempts, all with varying outcomes, but no clear resolution.

cc @lukehoban @stack72 @CyrusNajmabadi 
---

For starters, these are all the examples/tests we have.

There are 6 core examples:
1. `examples/cluster`
1. `examples/fargate`
1. `examples/nodegroup`
1. `examples/managed-nodegroups`
1. `examples/storage-classes`
1. `examples/tags`

There are 4 `with_update` core examples:
1. `examples/cluster`
1. `examples/nodegroup`
1. `examples/storage-classes`
1. `examples/tags`

There are 6 regression tests currently, and 2 new ones in the pipeline per https://github.com/pulumi/pulumi-eks/pull/302 :
1. `examples/tests/awsx-network-and-subnetIds`
1. `examples/tests/migrate-nodegroups`
1. `examples/tests/nodegroup-options`
1. `examples/tests/replace-cluster-add-subnets`
1. `examples/tests/replace-secgroup`
1. `examples/tests/tag-input-types`

and

1. `examples/tests/managed-ng-aws-auth`
1. `examples/tests/managed-ng-missing-role`

---

Fix attempts and outcomes:

1. [Commit](https://github.com/pulumi/pulumi-eks/commit/f9812efa407bb6c00b8937d69addada0c6577bde) | [Travis Run](https://travis-ci.com/pulumi/pulumi-eks/builds/143834557) | PASSED:
    - Ran a subset of tests
    - Enabled Logging in the `getRouteTables()` func of `nodegroup.ts` and did not encounter the `getRouteTables()` issue we've been seeing per [comment](https://github.com/pulumi/pulumi-eks/issues/300#issuecomment-571234664) above.
    - `PULUMI_TEST_DEBUG_LOG_LEVEL=2 && PULUMI_TEST_DEBUG_UPDATES=true`
1. [Commit](https://github.com/pulumi/pulumi-eks/commit/9ff9c6bd17806803ed2890070c7d4fcca669d0ef) | [Travis Run](https://travis-ci.com/pulumi/pulumi-eks/builds/144324952) | FAILED after 2 hour max test duration limit with `The job exceeded the maximum time limit for jobs, and has been terminated.`
    - Ran all tests
    - Enabled Logging in the `getRouteTables()` func of `nodegroup.ts` and did not encounter the `getRouteTables()` issue we've been seeing per [comment](https://github.com/pulumi/pulumi-eks/issues/300#issuecomment-571234664) above.
    - No PULUMI_TEST_DEBUG env-vars were enabled.
1. [Commit](https://github.com/pulumi/pulumi-eks/commit/2fd926c6e341f5b0a9722ed43b0ed4a080db898b) | [Travis Run](https://travis-ci.com/pulumi/pulumi-eks/builds/144337175) | FAILED with `No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.`:
    - Ran all tests
    - Removed debug logging in the `getRouteTables()` func of `nodegroup.ts`.
    - No PULUMI_TEST_DEBUG env-vars were enabled.
1. Last handful of Cron's ([#1306](https://travis-ci.com/pulumi/pulumi-eks/builds/144265374) and [#1304](https://travis-ci.com/pulumi/pulumi-eks/builds/144038567)) on last working commit / master  | FAILED with `No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.`
    -  `PULUMI_TEST_DEBUG_LOG_LEVEL=9 && PULUMI_TEST_DEBUG_UPDATES=true`, and
    -  `PULUMI_TEST_DEBUG_LOG_LEVEL=2 && PULUMI_TEST_DEBUG_UPDATES=true`
1. Re-ran last working commit / master | [Travis](https://travis-ci.com/pulumi/pulumi-eks/builds/144095685) | FAILED with `No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.` 
    -   No PULUMI_TEST_DEBUG env-vars were enabled.

---

In summary:

- The `getRouteTable: Your query returned no results` error has not been seen in about 1 week on Travis or locally. Instead we've seen tests stall out midway through with `No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself` across cron jobs and fix attempts.
- Enabling or not enabling the `PULUMI_TEST_DEBUG` does not seem to matter to a) tests running completely or b) helping them to pass
- The only test that did pass had only a subset of tests enabled. Given that `pulumi/examples` is ~90 tests, in theory there is no reason why all 16 current test of `pulumi/eks` cannot be all enabled.
- The only possible theory that's leading to such variance in error and lack of job completion is Travis, either due to network saturation, resource starvation, or spurious bugs that have been visibly noticed for a while now in both the UI and on job runs.

**Comment by CyrusNajmabadi** [comment]
Another issue:
https://travis-ci.com/pulumi/examples/builds/145618653

```
[ com/pulumi/examples/aws-ts-eks ]     Error: invocation of kubernetes:yaml:decode returned an error: unable to load Kubernetes client configuration from kubeconfig file: invalid configuration: no configuration has been provided

[ com/pulumi/examples/aws-ts-eks ]         at monitor.invoke (/tmp/p-it-travis-job-aws-ts-eks-c85be5f0-486422416/node_modules/@pulumi/pulumi/runtime/invoke.js:172:33)

[ com/pulumi/examples/aws-ts-eks ]         at Object.onReceiveStatus (/tmp/p-it-travis-job-aws-ts-eks-c85be5f0-486422416/node_modules/grpc/src/client_interceptors.js:1210:9)
```

**Comment by metral** [comment]
> Error: invocation of kubernetes:yaml:decode returned an error: unable to load Kubernetes client configuration from kubeconfig file: invalid configuration: no configuration has been provided

Related https://github.com/pulumi/pulumi-kubernetes/issues/958

**Comment by lblackstone** [comment]
The `yaml:decode` error was fixed in the 1.4.5 release of the k8s provider.

**Comment by metral** [comment]
> The yaml:decode error was fixed in the 1.4.5 release of the k8s provider.

Confirmed that i have not seen this today in EKS CI.

Still debugging rest of the issues we're seeing today:

- Travis runtime OOM'ing
- https://github.com/pulumi/pulumi/issues/3182
